src/components/splitting.rs

```diff
@@ -185,43 +189,27 @@ impl Component for Split {
 mod tests {
     use std::iter;
 
-    use derive_more::AsRef;
-
     use super::Split;
     use super::SplitKind;
     use crate::components::Echo;
     use crate::Component;
     use crate::Direction;
     use crate::DrawMode;
     use crate::Line;
-    use crate::State;
-
-    #[derive(AsRef, Debug)]
-    struct Echo1(Vec<Line>);
-    #[derive(AsRef, Debug)]
-    struct Echo2(Vec<Line>);
-
-    #[derive(AsRef, Debug)]
-    struct Echo3(Vec<Line>);
-
-    fn make_splitter(kind: SplitKind, dimension: Direction) -> Split {
-        Split::new(
-            vec![
-                Box::new(Echo::<Echo1>::new(false)),
-                Box::new(Echo::<Echo2>::new(false)),
-            ],
-            dimension,
-            kind,
-        )
-    }
 
     mod horizontal {
         use super::*;
         use crate::Dimensions;
         #[test]
         fn test_adaptive() {
-            let splitter = make_splitter(SplitKind::Adaptive, Direction::Horizontal);
-            let mut state = State::new();
+            let splitter = Split::new(
+                vec![
+                    Box::new(Echo::new_with_indices(false, (0, 3))),
```

Comment:
Example usage of `new_with_indices`. Notice how these indices correspond with `left_msg` and `right_msg`. These two line vecs get concatenated and flattened into a single line vec later on.